web/src/main/java/org/springframework/security/web/util/matcher/RequestMatcher.java

```diff
@@ -33,4 +37,61 @@
 	 */
 	boolean matches(HttpServletRequest request);
 
+	/**
+	 * @since 5.2
+	 */
+	default MatchResult matcher(HttpServletRequest request) {
+		boolean match = matches(request);
+		return new MatchResult(match, Collections.emptyMap());
+	}
+
+	/**
+	 * The result of matching
+	 */
+	class MatchResult {
+		private final boolean match;
+		private final Map<String, String> variables;
+
+		MatchResult(boolean match, Map<String, String> variables) {
+			this.match = match;
+			this.variables = variables;
+		}
+
+		public boolean isMatch() {
```

Comment:
Please add some JavaDoc for these two public methods.

Reply:
Completed